testplan/runners/pools/base.py

```diff
@@ -447,7 +473,9 @@ def _handle_taskresults(self, worker, request, response):
                 "De-assign {} from {}".format(task_result.task, worker)
             )
 
-            if self.should_reschedule(self, task_result):
+            if self.should_reschedule(self, task_result) or self.should_rerun(
+                task_result
+            ):
                 if self.task_assign_cnt[uid] >= self.cfg.task_retries_limit:
```

Comment:
retry: if worker dies (no result returned, detected by monitor), reschedule the task to pool - subject to pool.task_retries_limit. retry is currently enabled by default.

rerun: if task result is fail and user specifies so, reschedule the task to pool - subject to task.max_rerun. rerun can be controlled by user.

I think we shall keep these 2 and their limits separate